common/core/src/main/java/com/vmware/connectors/common/utils/Async.java

```diff
@@ -57,8 +65,8 @@ public void onSuccess(T result) {
             boolean same = callerContext == callbackContext; // NOPMD I really do want identity
             if (!same) {
                 callbackContext.putAll(callerContext);
-                if (mdcMap != null) {
-                    MDC.setContextMap(mdcMap);
+                if (!CollectionUtils.isEmpty(this.mdcMap)) {
```

Comment:
I was ```CollectionUtils.isEmpty``` to check if it is ```null``` and if it is ```isEmpty```. Since we are clearing the ```MDC```, I have reverted it to ```this.mdcMap != null```.